flink/v2.0/flink/src/main/java/org/apache/iceberg/flink/sink/IcebergFilesCommitter.java

```diff
@@ -138,6 +141,12 @@ class IcebergFilesCommitter extends AbstractStreamOperator<Void>
     this.spec = spec;
   }
 
+  @Override
+  protected void setup(
+      StreamTask<?, ?> containingTask, StreamConfig config, Output<StreamRecord<Void>> output) {
+    super.setup(containingTask, config, output);
+  }
+
```

Comment:
Why do we need this override?

Reply:
I forgot to add a comment here, but the reason is the same as in https://github.com/apache/iceberg/pull/12527/files/7eecf7fa428efbc7dd958d47e8cd74657d722541..35078bdc25647849afb06c58cdfbc5dda0c61830#diff-65fd76701c7646960de606de2cfe6ab6ba5f2eceb0ac66ca25ad321180127998R96

Changed.

Reply:
The reason is that the factory calls the setup method and the visibility has changed which requires an override here.

Reply:
you are saying `AbstractStreamOperator` changed the scope of this method from public to protected. can you share which operator factory still calls this method?

Reply:
It is called from here: https://github.com/apache/iceberg/blob/710276587d5f831f562f7c04abb7723c8b1824b0/flink/v2.0/flink/src/main/java/org/apache/iceberg/flink/source/StreamingReaderOperator.java#L235

Reply:
that was `StreamingReadOperator`. this is `IcebergFilesCommitter`. don't quite see the connection here.

Reply:
True, that was an example for a different operator. Generally, calling setup is required for all operators to initialize the task, its config, and its output. Flink moved away from an explicit call to calling this in the constructor of the base class: https://github.com/apache/flink/blob/b0607a15e62b664d15efbda0b0e991f72e45a467/flink-runtime/src/main/java/org/apache/flink/streaming/api/operators/AbstractStreamOperator.java#L166 We are not calling the super constructor, so we need to manually call the setup method. 

Looking at the code base, Flink 2.0 seems to favor creating a `StreamOperatorFactory`. I've updated the code to use a factory and call the super constructor instead. The setup method overrides are now removed.